source/common/http/conn_manager_impl.cc

```diff
@@ -446,6 +455,13 @@ Network::FilterStatus ConnectionManagerImpl::onData(Buffer::Instance& data, bool
       stats_.named_.downstream_cx_protocol_error_.inc();
       handleCodecError(status.message());
       return Network::FilterStatus::StopIteration;
+    } else if (isEnvoyOverloadError(status)) {
+      // The other codecs aren't wired to send this status.
+      ASSERT(codec_->protocol() < Protocol::Http2,
```

Comment:
Would anything misbehave if H2 or H3 sent this status? (Other than this ASSERT failing, of course :>)

Reply:
Given that, I *might* be inclined to remove this ASSERT() since it seems like things would work as expected. On the other hand, since we're not sending this status from other codecs, leaving it here is fine.